src/clients/client.ts

```diff
@@ -172,8 +167,11 @@ class Client {
     try {
       await this.deleteIndex(uid);
       return true;
-    } catch (e: any) {
-      if (e.code === ErrorStatusCode.INDEX_NOT_FOUND) {
+    } catch (e) {
+      if (
+        (e as { code?: ErrorStatusCode }).code ===
+        ErrorStatusCode.INDEX_NOT_FOUND
+      ) {
```

Comment:
Use `instanceof` instead of casting, with the appropriate error.

Reply:
the problem here is that this will throw a MeiliSearchErrorResponse, but that is not a class rather a type so I can't use `instanceof`
![meili_error](https://github.com/user-attachments/assets/5cfc1b25-a7b3-4f58-9e4c-91a0cf9f1cbe)

Reply:
It's `MeiliSearchApiError`, it cannot possibly throw the type of object you are referring to on its own. The code is actually wrong, when I updated the errors in one of my PRs I forgot to change that. It should be checking `error.cause.code`, not `error.code`.

Reply:
I understand. Updated it accordingly